meshtastic/ble_interface.py

```
@@ -34,58 +33,86 @@ class BLEError(Exception):
+            logger.warning("Event loop not running, can't run coroutine")
+            return None
 
-        # We MUST run atexit (if we can) because otherwise (at least on linux) the BLE device is not disconnected
-        # and future connection attempts will fail.  (BlueZ kinda sucks)
-        # Note: the on disconnected callback will call our self.close which will make us nicely wait for threads to exit
-        self._exit_handler = atexit.register(self.client.disconnect)
+        try:
+            future = asyncio.run_coroutine_threadsafe(coro, self._event_loop)
+            return future.result(timeout)
+        except asyncio.TimeoutError as e:
+            logger.error(f"Coroutine execution timed out: {e}")
```

Ruff (0.13.1)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)